File: solutions/2657/solution.java
class Solution {
    public int[] findThePrefixCommonArray(int[] A, int[] B) {
        int size = A.length;
        int[] freq = new int[size+1];
        int[] ans  = new int[size];
        int count = 0;
        for(int i=0;i<size;i++){
            freq[A[i]]++;
            if(freq[A[i]] == 2) count++;
            freq[B[i]]++;
            if(freq[B[i]] == 2) count++;
            ans[i] = count;
        }
        return ans;
    }
}

File: solutions/2657/solution.py
class Solution:
    def findThePrefixCommonArray(self, A: List[int], B: List[int]) -> List[int]:
        size = len(A)
        freq = [0]*(size+1)
        ans = [0]*size
        count = 0
        for i in range(size):
            freq[A[i]]+=1
            if freq[A[i]] == 2:
                count+=1
            freq[B[i]]+=1
            if freq[B[i]] == 2:
                count+=1
            ans[i] = count
        return ans